### ISSUE TITLE
Prevent from navigating away when there are pending changes to save

### ISSUE THREAD

**Issue by raulgomis** [sdorra/angular-dashboard-framework#147]
Hi,

It would be great that there is a warning when the user is navigating away trying to avoid the pending changes that he/she needs to save.

There is one possible good solution to achieve this:

At first the dashboard sets a variable dirty = false. And when any component is changed / created, this variable should be set to true. Then, this variable would be set to false again the all the changes in the dashboard are saved. 
We should prevent: any angular route navigating and any browser navigating / refreshing 

Kind of javascript pseudocode:
```javascript
var dirty = false;

//prevent browser navigating away
window.onbeforeunload = function() {
    return dirty ? "If you leave this page you will lose your unsaved changes." : null;
}
//TODO: prevent also angular navigation
```
What do you think @sdorra?

**Comment by sdorra** [comment]
I think this would be a good addition. But i think it is not so easy, because we do not know which routing provider is used with adf. Is the onbeforeunload event is fired on a spa navigation change?

**Comment by raulgomis** [comment]
Unfortunately, I don't think you could detect and SPA change with the **onbeforeunload** event. We could use the $locationChangeStart event to detect a location change, which is not coupled with neither ng-route nor ui-router (it's in the angular's core). It would be something like:

```javascript
$scope.$on('$locationChangeStart', function( event ) {
    var answer = confirm("Are you sure you want to leave this page?")
    if (!answer) {
        event.preventDefault();
    }
});
```

Moreover, I'm not sure, but I don't think that angular location would listen to the 'close page / tab' event. So, we should implement both **onbeforeunload** and **$locationChangeStart** events to work properly.

**Comment by sdorra** [comment]
@raulgomis sounds good, but i think we should add an option to enable/disable this behaviour.

**Comment by raulgomis** [comment]
yeah! definitely! also, we should do it configurable in order to let users change the message for i18n, etc.

**Comment by eppak** [comment]
Any news of this feature? (cannot find track on sources) Can i ask where to put it (if not yet merged)? I tried to put it in my main module (app side) and it fire on page load (not unload).

Thanks
